common/src/main/java/com/tc/net/core/BufferManager.java

```diff
@@ -27,12 +27,40 @@
  * @author Ludovic Orban
  */
 public interface BufferManager {
+  /**
+   * Called immediately after a call to recvToBuffer(). It will be called again if it returns a non-zero value.
+   * It is called from the connection's read thread.
+   * This method should write data into the supplied ByteBuffer.
+   * @param dest the ByteBuffer into which data should be written
+   * @return the number of bytes written
+   */
   int forwardFromReadBuffer(ByteBuffer dest);
 
+  /**
+   * Called when there is data to write to the channel and the channel is accepting writes.
+   * It is called from the connection's write thread.
```

Comment:
`It is called from the connection's write thread.` is an implementation detail of the comm stack.